diptest/game_test.go

```diff
@@ -2,12 +2,132 @@ package diptest
+	if g[0].Desc != "c" {
+		t.Errorf("got %q, wanted 'c'", g[0].Desc)
+	}
+	if g[1].Desc != "b" {
+		t.Errorf("got %q, wanted 'b'", g[1].Desc)
+	}
+	if g[2].Desc != "a" {
+		t.Errorf("got %q, wanted 'a'", g[2].Desc)
+	}
+}
+
+var (
+	uniqueMaxHated uint64 = uint64(time.Now().UnixNano() / 1000000000)
+)
+
+func TestGameMerging(t *testing.T) {
+	maxHated := atomic.AddUint64(&uniqueMaxHated, 1)
+	gameDesc := String("test-game")
+	env := NewEnv().SetUID(String("fake"))
+	t.Run("CreateGame", func(t *testing.T) {
+		env.GetRoute(game.IndexRoute).Success().
+			Follow("create-game", "Links").
+			Body(map[string]interface{}{
+				"Variant":            "Classical",
+				"Desc":               gameDesc,
+				"MaxHated":           float64(maxHated),
+				"PhaseLengthMinutes": time.Duration(60),
+			}).Success().
+			AssertEq(gameDesc, "Properties", "Desc")
+
+		env.GetRoute(game.ListMyStagingGamesRoute).Success().
+			Find(gameDesc, []string{"Properties"}, []string{"Properties", "Desc"})
+	})
+
+	t.Run("VerifySelfGameNoMerge", func(t *testing.T) {
+		gameDesc2 := String("test-game")
+		env.GetRoute(game.IndexRoute).Success().
+			Follow("create-game", "Links").
+			Body(map[string]interface{}{
+				"Variant":            "Classical",
+				"MaxHated":           float64(maxHated),
+				"Desc":               gameDesc2,
+				"PhaseLengthMinutes": time.Duration(60),
+			}).Success().
+			AssertEq(gameDesc2, "Properties", "Desc")
+		env.GetRoute(game.ListMyStagingGamesRoute).Success().
+			Find(gameDesc2, []string{"Properties"}, []string{"Properties", "Desc"})
+	})
+
+	t.Run("VerifyNoMergeGameNoMerge", func(t *testing.T) {
+		env2 := NewEnv().SetUID(String("fake"))
+		gameDesc3 := String("test-game")
+		env2.GetRoute(game.IndexRoute).Success().
+			Follow("create-game", "Links").
+			Body(map[string]interface{}{
+				"Variant":            "Classical",
+				"Desc":               gameDesc3,
+				"MaxHated":           float64(maxHated),
+				"NoMerge":            true,
+				"PhaseLengthMinutes": time.Duration(60),
+			}).Success().
+			AssertEq(gameDesc3, "Properties", "Desc")
+		env2.GetRoute(game.ListMyStagingGamesRoute).Success().
+			Find(gameDesc3, []string{"Properties"}, []string{"Properties", "Desc"})
+	})
+
+	t.Run("VerifyDifferentVariantGameNoMerge", func(t *testing.T) {
+		env3 := NewEnv().SetUID(String("fake"))
+		gameDesc4 := String("test-game")
+		env3.GetRoute(game.IndexRoute).Success().
+			Follow("create-game", "Links").
+			Body(map[string]interface{}{
+				"Variant":            "Fleet Rome",
+				"MaxHated":           float64(maxHated),
+				"Desc":               gameDesc4,
+				"PhaseLengthMinutes": time.Duration(60),
+			}).Success().
+			AssertEq(gameDesc4, "Properties", "Desc")
+		env3.GetRoute(game.ListMyStagingGamesRoute).Success().
+			Find(gameDesc4, []string{"Properties"}, []string{"Properties", "Desc"})
+	})
+
+	t.Run("VerifyEqualGameMerge", func(t *testing.T) {
+		env4 := NewEnv().SetUID(String("fake"))
+		gameDesc5 := String("test-game")
```

Comment:
Is it worth changing the game name?

Reply:
The game is actually changed - String("test-game") will create a unique string each time, containing (for debugability) "test-game".

Reply:
Ah ok. The method name 'String' could be clearer then! I remember it now from seeing the fake users after running the existing tests.

Arguably the game names in the other tests should match the first game name. I guess that makes debugging harder though.

Reply:
Yeah - the name 'String' was mostly for historic reasons, another project happened to have that name to generate test-strings...

Since the merging code ignores the game name, I didn't think it was important. And the testing would be harder, since I verify merge or no merge by looking for games with particular names.